worlds/mm2/rom.py

```diff
@@ -0,0 +1,415 @@
+        patch.write_byte(0x2D3A0, color)  # Change white to a dark gray, Picopico-kun/Boobeam Trap
+        patch.write_byte(0x2D65F, color)  # Change white to a dark gray, Guts Tank
+        patch.write_byte(0x2DA94, color)  # Change white to a dark gray, Wily Machine
+        patch.write_byte(0x2DC97, color)  # Change white to a dark gray, Alien
+        patch.write_byte(0x2DD68, 0x10)  # Longer flash time, Alien kill
+        patch.write_bytes(0x2DF14, [0xEA, 0xEA, 0xEA, 0xEA, 0xEA, 0xEA])  # Reduce final Alien flash to 1 big flash
+        patch.write_byte(0x34132, 0x08)  # Longer flash time, Stage Select
+
+        if world.options.reduce_flashing.value == ReduceFlashing.option_full:
+            # reduce color of stage flashing
+            patch.write_bytes(0x344C9, [0x2D, 0x10, 0x00, 0x2D,
+                                        0x0F, 0x10, 0x2D, 0x00,
+                                        0x0F, 0x10, 0x2D, 0x00,
+                                        0x0F, 0x10, 0x2D, 0x00,
+                                        0x2D, 0x10, 0x2D, 0x00,
+                                        0x0F, 0x10, 0x2D, 0x00,
+                                        0x0F, 0x10, 0x2D, 0x00,
+                                        0x0F, 0x10, 0x2D, 0x00])
+            # remove wily castle flash
+            patch.write_byte(0x3596D, 0x0F)
+
+        if speed != -1:
+            patch.write_byte(0xFE01, speed)  # Bubble Man Stage
+            patch.write_byte(0x1BE01, speed)  # Metal Man Stage
+
+    if world.options.random_music:
+        if world.options.random_music == RandomMusic.option_none:
+            pool = [0xFF] * 13
+            # A couple of additional mutes we want here
+            patch.write_byte(0x37819, 0xFF)  # Credits
+            patch.write_byte(0x378A4, 0xFF)  # Credits #2
+            # patch.write_byte(0x37149, 0xFF)  # Game Over Jingle
+            # patch.write_byte(0x341BA, 0xFF)  # Robot Master Jingle
+            # patch.write_byte(0x2E0B4, 0xFF)  # Robot Master Defeated
+            # patch.write_byte(0x35B78, 0xFF)  # Wily Castle
+            # patch.write_byte(0x2DFA5, 0xFF)  # Wily Defeated
+
+        elif world.options.random_music == RandomMusic.option_shuffled:
+            pool = [0, 1, 2, 3, 4, 5, 6, 7, 8, 8, 9, 9, 9, 0x10, 0xC, 0xB, 0x17, 0x13, 0xE, 0xD]
+            world.random.shuffle(pool)
+        else:
+            pool = world.random.choices(range(10), k=13)
+        patch.write_bytes(0x381E0, pool[:13])
+        patch.write_byte(0x36318, pool[13])  # Game Start
+        patch.write_byte(0x37181, pool[13])  # Game Over
+        patch.write_byte(0x360AE, pool[14])  # RBM Select
+        patch.write_byte(0x39005, pool[15])  # Robot Master Battle
+        patch.write_byte(get_equipped_sound_ptr + 1, pool[16])  # Get Equipped, we actually hook this already lmao
+        patch.write_byte(0x3775A, pool[17])  # Epilogue
+        patch.write_byte(0x36089, pool[18])  # Intro
+        patch.write_byte(0x3)
```

Comment:
Here, there is a missing arguments.

Reply:
Scipio's comments made me distracted from finishing this lol.